### PR TITLE
MM-35332 Cypress/E2E: Not extending session when setting is disabled

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Related PR - https://github.com/mattermost/mattermost-server/pull/17581

Test cases:
- [MM-T4045](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4045) - session via Email login
- [MM-T4046](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4046) - session via LDAP login
- [MM-T4047](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4047) - session via SAML login

Note:
- Separated into its own spec file since they require different auth mechanism and to make it easier to move individually into production test.
- Test will fail if your server doesn't include the recent fix - https://github.com/mattermost/mattermost-server/commit/a668479992047548e6d40eed4e5cc1d408161447

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35332

#### Screenshots
Screen Shot 2021-05-08 at 6 44 54 AM
Screen Shot 2021-05-08 at 6 53 27 AM
Screen Shot 2021-05-08 at 6 58 10 AM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by josephbaylon — APPROVED** [review]
@saturninoabril I'm not able to run the saml case because I'm not sure how to spawn a keycloak server. I'm approving anyway since the other 2 passed for me locally and assuming the saml passed for you locally.

**Comment by saturninoabril** [comment]
Thanks Joseph and no worries. To run keycloak and other service like openldap/elasticsearch, you may add the docker image thru environment variable:
```bash
export ENABLED_DOCKER_SERVICES="postgres inbucket keycloak"
```
Do server restart and you'll find `mattermost-saml` service up and running. Also, you'll find your local docker setup at `/mattermost-server/docker-compose.makefile.yml`.

See [this line from server](https://github.com/mattermost/mattermost-server/blob/master/config.mk#L13) for more info.

**Review by wiggin77 — APPROVED** [review]
This looks great @saturninoabril  :+1:

**Comment by josephbaylon** [comment]
@saturninoabril I enabled the keycloak but it's now complaining about missing fixture

Screen Shot 2021-05-12 at 12 29 16 PM

File: e2e/cypress/integration/enterprise/extend_session/not_extended_when_disabled/with_saml_login_spec.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Requires openldap and keycloak running
+// Requires keycloak certificate at fixtures folder
+// - copy ./mattermost-server/build/docker/keycloak/keycloak.crt to ./mattermost-webapp/e2e/cypress/fixtures/keycloak.crt
+// Requires Cypress' chromeWebSecurity to be false
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/extend_session/not_extended_when_disabled/with_saml_login_spec.js:13]: @josephbaylon You'll need to manually copy the [keycloak.crt from server repo](https://github.com/mattermost/mattermost-server/blob/master/build/docker/keycloak/keycloak.crt) into the fixtures folder. Let me know if we need to modify the instructions above.

Comment (josephbaylon) [reply e2e/cypress/integration/enterprise/extend_session/not_extended_when_disabled/with_saml_login_spec.js:13]: My bad. I didn't notice the instructions.

**Pull request merged**